src/commands/up.rs

```diff
@@ -87,6 +87,14 @@ pub struct UpCommand {
     #[clap(long = "temp")]
     pub tmp: Option<PathBuf>,
 
+    /// For local apps with directory mounts and no excluded files, mount them directly instead of using a temporary
+    /// directory.
+    ///
+    /// This allows you to update the assets on the host filesystem such that the updates are visible to the guest
+    /// without a restart.  This will raise an error if used with bindle apps or apps using file patterns.
+    #[clap(long, takes_value = false)]
+    pub direct_mounts: bool,
```

Comment:
Does it make sense to provide `conflicts_with` for `--temp` and `--bindle` opts?

Reply:
Good idea.  I originally had it conflict with `--temp`, but then realized that Spin uses the temporary directory for other things besides file and directory mounts, so it's still useful to specify `--temp` and `--direct-mounts` simultaneously.  I agree that it should conflict with the bindle opts, though.

Reply:
Oh, right, sorry, I missed that.

I misunderstood the term "static assets" in the opt description:

https://github.com/fermyon/spin/blob/main/src/commands/up.rs#L86-L88

Reply:
Yeah, I think that description is misleading -- I was also confused. I suspect it was originally just for static assets, but then ended up being used for other things.